src/id/westmanga/src/eu/kanade/tachiyomi/extension/id/westmanga/WestManga.kt

```diff
@@ -1,53 +1,213 @@
+            author = data.author
+            status = when (data.status) {
+                "ongoing" -> SManga.ONGOING
+                "completed" -> SManga.COMPLETED
+                "hiatus" -> SManga.ON_HIATUS
+                else -> SManga.UNKNOWN
             }
-            val genres = seriesDetails.select(seriesGenreSelector).map { it.text() }.toMutableList()
-            // Add series type (manga/manhwa/manhua/other) to genre
-            seriesDetails.selectFirst(seriesTypeSelector)?.ownText().takeIf { it.isNullOrBlank().not() }?.let { genres.add(it) }
-            genre = genres.map { genre ->
-                genre.lowercase(Locale.forLanguageTag(lang)).replaceFirstChar { char ->
-                    if (char.isLowerCase()) {
-                        char.titlecase(Locale.forLanguageTag(lang))
-                    } else {
-                        char.toString()
-                    }
+            genre = buildList {
+                when (data.country) {
+                    "JP" -> add("Manga")
+                    "CN" -> add("Manhua")
+                    "KR" -> add("Manhwa")
+                }
+                if (data.color) {
+                    add("Colored")
                 }
+                data.genres.forEach { add(it.name) }
+            }.joinToString()
+            description = buildString {
+                append(
+                    Jsoup.parseBodyFragment(data.synopsis).wholeText(),
+                )
+                append("\n\n")
+                append("Alternative Name: ")
+                append(data.alternativeName.trim())
+            }
+        }
+    }
+
+    override fun chapterListRequest(manga: SManga) =
+        mangaDetailsRequest(manga)
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val data = response.parseAs<Data<Manga>>().data
+
+        return data.chapters.map {
+            SChapter.create().apply {
+                url = "/${it.slug}/"
+                name = "Chapter ${it.number}"
+                date_upload = it.updatedAt.time * 1000
             }
-                .joinToString { it.trim() }
+        }
+    }
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        val slug = "$baseUrl${chapter.url}".toHttpUrl().pathSegments[0]
+        val url = baseUrl.toHttpUrl().newBuilder().apply {
+            addPathSegment("api")
+            addPathSegment("v")
+            addPathSegment(slug)
+        }.build()
 
-            status = seriesDetails.selectFirst(seriesStatusSelector)?.text().parseStatus()
-            thumbnail_url = seriesDetails.select(seriesThumbnailSelector).imgAttr()
+        return apiRequest(url)
+    }
+
+    override fun getChapterUrl(chapter: SChapter): String {
+        val slug = "$baseUrl${chapter.url}".toHttpUrl().pathSegments[0]
+        val url = baseUrl.toHttpUrl().newBuilder().apply {
+            addPathSegment("view")
+            addPathSegment(slug)
+        }.build()
```

Comment:
I don't think you need `.apply` here. Static string can also be moved:
```suggestion
        val url = "$baseUrl/view".toHttpUrl().newBuilder()
            .addPathSegment(slug)
            .build()
```
or:
```suggestion
        val url = baseUrl.toHttpUrl().newBuilder()
            .addPathSegment("view")
            .addPathSegment(slug)
            .build()
```

The same can be said for `pageListRequest()`.